### PR TITLE
PATCH and DELETE methods for folders

### LINKED ISSUE #97: Add support for adding metadata object folders
#### Proposed solution

Users should be able to collect metadata objects into folders. Folders are mainly views to how objects are linked together, so at first we could have following:
- endpoint with crud methods for folder: 
   - /folders/ GET --> return all folders, POST --> add new
   - /folders/{folder_id} GET --> return one folder, PUT --> replace one folder, PATCH --> partially update one folder, DELETE --> delete folder
- support for folders in database

Suggestion for folder-json structure: 
```
{
  id: "Mongodb generated Id", 
  metadata_objects: [
{ "accessionId": "EGA12356", 
  "schema": "Study" }, 
{ "accessionId": "EGA12356339", 
  "schema": "Sample" }, 
{ "accessionId": "EGA12356398", 
  "schema": "Experiment" }, 
]
}
```

#### DoD (Definition of Done)
Folder endpoints are implemented 

#### Testing
Unit and integration tests

### PR DESCRIPTION
**Pull request by csc-jm**
### Description

PR includes handling for PATCH and DELETE requests of folders (PUT was there as well for a moment but deemed unnecessary). Also PR tidies things up with the creation of `FolderOperator` in `operators.py` and moving DB operations there as opposed to having them in `handlers.py`. The PATCH method for folders also utilizes `jsonpatch` python library [(http://jsonpatch.com/)](http://jsonpatch.com/).

### Related issues

Finally fixes #97. (hopefully)

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- added PATCH method for folders
- added DELETE method for folders
- altered methods in `DBService` class to work with folder operations
- moved all JSON data checks in `handlers.py` into its own helper method
- added new `FolderOperator` class and moved DB operations from `handlers.py` there
- added some unit tests
- fixed some typos found along the way

### Testing

- [x] Unit Tests
- [x] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: metadata_backend/api/handlers.py [hunk 1]
```diff
@@ -115,6 +116,23 @@ def get_page_param(param_name: str, default: int) -> int:
...
+    async def _get_data(self, req: Request) -> Dict:
+        """Get the data content from a request.
+
+        :param req: POST/PUT/PATCH request
+        :raises: HTTPBadRequest if request does not have proper JSON data
+        :returns: JSON content of the request
+        """
+        try:
+            content = await req.json()
+            # if content['type']
```
Comment (blankdots) [inline metadata_backend/api/handlers.py:128]: probably remove this ?

File: metadata_backend/database/db_service.py [hunk 2]
```diff
@@ -73,25 +73,26 @@ async def create(self, collection: str, document: Dict) -> bool:
...
 
     @auto_reconnect
-    async def exists(self, collection: str, accession_id: str) -> bool:
-        """Check object exists by its accessionId.
+    async def exists(self, collection: str, id: str) -> bool:
+        """Check object or folder exists by its generated ID.
 
         :param collection: Collection where document should be searched from
-        :param accession_id: Accession id of the document to be searched
+        :param accession_id: ID of the document or folder to be searched
```
Comment (blankdots) [inline metadata_backend/database/db_service.py:80]:
```suggestion
        :param accession_id: ID of the object or folder to be searched
```

File: metadata_backend/database/db_service.py [hunk 3]
```diff
@@ -73,25 +73,26 @@ async def create(self, collection: str, document: Dict) -> bool:
...
         return True if exists else False
 
     @auto_reconnect
     async def read(self, collection: str, id: str) -> Dict:
-        """Find object by its accessionId.
+        """Find object or folder by its generated ID.
 
         :param collection: Collection where document should be searched from
-        :param id: Accession id of the document or folder id of the folder
+        :param id: ID of the document or folder to be searched
```
Comment (blankdots) [inline metadata_backend/database/db_service.py:95]:
```suggestion
        :param id: ID of the object or folder to be searched
```

File: metadata_backend/database/db_service.py (hunk 2 above)
Comment (otahontas) [reply metadata_backend/database/db_service.py:80]: this should be actually just `:param id`

File: metadata_backend/api/handlers.py (hunk 1 above)
Comment (csc-jm) [reply metadata_backend/api/handlers.py]: No idea what that was doing there, so a good catch 👍

**Pull request merged**